examples/tree_2d_dgsem/elixir_euler_riemannproblem_quadrants_amr.jl

```diff
@@ -0,0 +1,159 @@
+using OrdinaryDiffEqSSPRK
+using Trixi
+
+###############################################################################
+## Semidiscretization of the compressible Euler equations
+
+equations = CompressibleEulerEquations2D(1.4)
+
+# Variant of the 4-quadrant Riemann problem considered in 
+# - Carsten W. Schulz-Rinne:
+#   Classification of the Riemann Problem for Two-Dimensional Gas Dynamics
+#   https://doi.org/10.1137/0524006
+# and 
+# - Carsten W. Schulz-Rinne, James P. Collins, and Harland M. Glaz
+#   Numerical Solution of the Riemann Problem for Two-Dimensional Gas Dynamics
+#   https://doi.org/10.1137/0914082
+function initial_condition_rp(x_, t, equations::CompressibleEulerEquations2D)
+    x, y = x_[1], x_[2]
+
+    if x >= 0.5 && y >= 0.5
+        rho, v1, v2, p = (0.5313, 0.0, 0.0, 0.4)
+    elseif x < 0.5 && y >= 0.5
+        rho, v1, v2, p = (1.0, 0.7276, 0.0, 1.0)
+    elseif x < 0.5 && y < 0.5
+        rho, v1, v2, p = (0.8, 0.0, 0.0, 1.0)
+    elseif x >= 0.5 && y < 0.5
+        rho, v1, v2, p = (1.0, 0.0, 0.7276, 1.0)
+    end
+
+    prim = SVector(rho, v1, v2, p)
+    return prim2cons(prim, equations)
+end
+initial_condition = initial_condition_rp
+
+# See Section 2.3 of the reference below for a discussion of robust
+# subsonic inflow/outflow boundary conditions.
+#
+# - Jan-Reneé Carlson (2011)
+#   Inflow/Outflow Boundary Conditions with Application to FUN3D.
+#   [NASA TM 20110022658](https://ntrs.nasa.gov/citations/20110022658)
+@inline function boundary_condition_subsonic(u_inner, orientation::Integer,
+                                             normal_direction, x, t,
+                                             surface_flux_function,
+                                             equations::CompressibleEulerEquations2D)
+    rho_loc, v1_loc, v2_loc, p_loc = cons2prim(u_inner, equations)
+
+    p_loc = pressure(initial_condition_rp(x, t, equations), equations)
+
+    prim = SVector(rho_loc, v1_loc, v2_loc, p_loc)
+    u_surface = prim2cons(prim, equations)
+
+    return Trixi.flux(u_surface, orientation, equations)
+end
+
+boundary_conditions = (x_neg = boundary_condition_subsonic,
+                       x_pos = boundary_condition_do_nothing,
```

Comment:
The results are clearly greatly, but do you know how this works?  E.g, `x_pos` is also subsonic.

Reply:
I provided an intuitive explanation taking the velocities into account.